cli/cli_app.go

```diff
@@ -500,6 +510,16 @@ func runTerraformInit(terragruntOptions *options.TerragruntOptions, terragruntCo
 	return runTerragruntWithConfig(initOptions, terragruntConfig, downloadSource)
 }
 
+// automatically run `init` for us when terraform says we should and continues to run the original command from terrgruntConfig
+// returns error if unable to run init (possibly because auto-init is false), or if the underlying command fails
+func autorunTerrafromInit (terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig, tfoutput string) error {
+	terragruntOptions.Logger.Printf("terraform returned %s, running init to fix it\n\n", tfoutput)
+	if err := runTerraformInit(terragruntOptions, terragruntConfig, nil); err != nil {
+			return err
+	}
+	return runTerragruntWithConfig(terragruntOptions, terragruntConfig, false)
```

Comment:
Does this end up re-running `init` further down? I just want to make sure we don't end up in an infinite retry loop here!